### ISSUE TITLE
TYPES with identical field names

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1936]
should give error,

```abap
TYPES: BEGIN OF main,
           foo TYPE i,
           foo TYPE i,
         END OF main.
```
